app/screens/channel/channel_nav_bar/channel_nav_bar.test.js

```diff
@@ -49,4 +53,74 @@ describe('ChannelNavBar', () => {
 
         expect(wrapper.state('permanentSidebar')).toBeDefined();
     });
+
+    test('drawerButtonVisible appears for android tablets', () => {
+        const wrapper = shallow(
+            <ChannelNavBar {...baseProps}/>,
+        );
+
+        DeviceTypes.IS_TABLET = true;
+        Platform.OS = 'android';
+
+        expect(wrapper.instance().drawerButtonVisible()).toBe(true);
+    });
+
+    test('drawerButtonVisible appears for android phones', () => {
+        const wrapper = shallow(
+            <ChannelNavBar {...baseProps}/>,
+        );
+
+        DeviceTypes.IS_TABLET = false;
+        Platform.OS = 'android';
+
+        expect(wrapper.instance().drawerButtonVisible()).toBe(true);
+    });
+
+    test('drawerButtonVisible appears for iOS phones', () => {
+        const wrapper = shallow(
+            <ChannelNavBar {...baseProps}/>,
+        );
+
+        DeviceTypes.IS_TABLET = false;
+        Platform.OS = 'ios';
+
+        expect(wrapper.instance().drawerButtonVisible()).toBe(true);
+    });
+
+    test('drawerButtonVisible appears for iOS tablets', () => {
+        const wrapper = shallow(
+            <ChannelNavBar {...baseProps}/>,
+        );
+
+        DeviceTypes.IS_TABLET = true;
```

Comment:
This case is the "default" state for iOS tablets  i.e  where `permanentSidebar` and `isSplitView` are not touched (left to their defaults). I agree that these should be explicitly mentioned.

Reply:
This case cleaned up in 39ecdc415ed395ab11d4c84d0bdfb2bec4c8ee83.

The subsequent tests 👇 handle the rest of the truth table when device is a tablet (`IS_TABLET = true`) and the other two switches are alternately set.

Reply:
You're correct in your analysis. The test **should** fail if the `||`s are omitted and passes only because of the setting of the subsequent conditions, which is now explicit after the recent commit. Hence the re-titling of the test spec.